Makefile

```diff
@@ -1,18 +1,37 @@
-.PHONY: dev clean e2e push
+.PHONY: clean dev dev-private-cluster e2e e2e-private-cluster publish-tag push push-private
+
+clean:
+	rm -rf devenv/state devenv/.terraform.lock.hcl devenv/.terraform devenv/terraform.tfstate devenv/terraform.tfstate.backup
 
 dev:
 	terraform --version
 	cd devenv && mkdir -p state && terraform init && terraform apply -auto-approve
 
-clean:
-	rm -rf devenv/state devenv/.terraform.lock.hcl devenv/.terraform devenv/terraform.tfstate devenv/terraform.tfstate.backup
+dev-private-cluster:
+	terraform --version
+	cd devenv && mkdir -p state && terraform init && terraform apply -auto-approve -var="private-dns=true"
+	cd devenv && /bin/sh scripts/deploy_addon_private_cluster.sh
 
 e2e:
 	go test -v --count=1 --tags=e2e ./e2e
 
-push:
+e2e-private-cluster:
+	# grab the image name and tag - that first gets determined by where and how we push it - so first need to build and push it - then export that name and registry and tag into the YAML
+	az acr login -n `cat devenv/state/registry.txt`
+	echo "$(shell cat devenv/state/registry.txt)/app-routing-operator-e2e:$(shell date +%s)" > devenv/state/e2e-image-tag.txt
+	docker build --platform=linux/amd64 -t `cat devenv/state/e2e-image-tag.txt` -f e2e/Dockerfile .
+	docker push `cat devenv/state/e2e-image-tag.txt`
+	cd devenv && /bin/sh scripts/deploy_private_e2e_tester.sh
```

Comment:
switching to /usr/bin/env instead of /bin/sh would make it more compatible with other shells users could have when develping